src/Polly.Shared/Caching/CacheEngine.cs

```diff
@@ -7,7 +7,7 @@ internal static partial class CacheEngine
     {
         internal static TResult Implementation<TResult>(
             ISyncCacheProvider<TResult> cacheProvider,
-            ITtlStrategy ttlStrategy,
+            ITtlStrategy<TResult> ttlStrategy,
```

Comment:
We need to make the async version of CachePolicy follow the same pattern, so I think we need to:

+ make the _same_ change on line 11 of of `CacheEngineAsync`, ie change the incoming ttl parameter to ` ITtlStrategy<TResult> ttlStrategy`
+ make consequent changes to `CachePolicyAsync.cs`, exactly same pattern as `CachePolicy.cs` already changed
+ consequent changes to `CacheTResultSyntaxAsync.cs`, exactly same pattern as for `CacheTResultSyntax.cs` already 

@awarrenlove Are you good to bring those changes in?  Basically just making the async version of the cache policy in line with the sync changes you have already done